specs/windows/windows_eventlog.table

```diff
@@ -0,0 +1,23 @@
+table_name("windows_eventlog")
+description("Table for querying all recorded Windows event logs.")
+schema([
+    Column("channel", TEXT, "Source or channel of the event", required=True),
+    Column("datetime", TEXT, "System time at which the event occurred"),
+    Column("task", INTEGER, "Task value associated with the event"),
+    Column("level", INTEGER, "Severity level associated with the event"),
+    Column("provider_name", TEXT, "Provider name of the event"),
+    Column("provider_guid", TEXT, "Provider guid of the event"),
+    Column("eventid", INTEGER, "Event ID of the event", additional=True),
+    Column("keywords", TEXT, "A bitmask of the keywords defined in the event"),
+    Column("data", TEXT, "Data associated with the event"),
+    Column("pid", INTEGER, "Process ID which emitted the event record", additional=True),
+    Column("tid", INTEGER, "Thread ID which emitted the event record"),
+    Column("time_range", TEXT, "System time to selectively filter the events", hidden=True, additional=True),
+    Column("timestamp", TEXT, "Timestamp to selectively filter the events", hidden=True, additional=True),
+    Column("xpath", TEXT, "The custom query to filter events", hidden=True, required=True),
```

Comment:
Does it make sense to support `xpath` here, vs building the expression based on the where clauses?

Reply:
IMO it makes sense to support both to provide flexibility for the user. We just need to work out the UX for when both are specified.

Reply:
Thanks @directionless and @zwass for the comments. The `xpath` feature is optional and it can be used to pass the complex query string which can't be generated with the where clauses. An example query could look like:

```
<QueryList>
  <Query Id="0">
    <Select Path="Security">*[System[band(Keywords,4503599627370496)] and EventData[Data[@Name = 'SubjectUserSid']]]</Select>
  </Query>
</QueryList>
```

Here the user can filter the events based on EventData. As @zwass mentioned, I added it to provide flexibility to the user and they can make more fine-grained queries if they wants.

Reply:
I like the ability to do fine grained filtering like this. But I don't like providing two filter mechanisms with unclear interactions. 

I'm not sure what I expect if I queried `select * from windows_eventlog where pid=123 and xpath=<whatever>` I'd probably expect them to be additive. (that's what the sql `and` means)

Reply:
Throwing an error and returning nothing might be okay too.

Reply:
I'd go with error. Seems risky to try to modify the xml provided by the user.

Reply:
The `xpath` filtering mechanism is exclusive and does not interact with the other approach but still, it can be used with some of the constraints. A trivial example could be:
```
select * from windows_eventlog where xpath = '<QueryList><Query Id="0"><Select Path="Application">*[System]</Select></Query></QueryList>' and pid = 1376;
```
It will query the events based on `xpath` but the SQLite will do the filtering based on `pid`.  This might not be the right approach to use the xpath but the query results will be consistent.

Reply:
I added a check and it will throw error msg if finds the constraints which should not be used with the `xpath`. The `xpath` also can't be used with itself and should be mutually exclusive.